src/Clide.Interfaces/Extensions/ServiceProviderExtensions.cs

```diff
@@ -119,39 +119,58 @@ internal static Guid GetPackageGuidOrThrow(this IServiceProvider owningPackage)
         return new Guid(guid.Value);
     }
 
+    static Guid GetPackageGuidOrThrow<TPackage>()
```

Comment:
Does it worth to add a condition over the TPackage to inherit from Package class

Reply:
The reality is that we only use the type to retrieve the guid from its attribute, so the actual type of the T isn't very relevant, and it's not casted or anything. Just for correctness maybe?